### PR TITLE
Docker compose

### PR DESCRIPTION
**Pull request by haffi96**
- Adding a way to easily run a distributed mite with metrics

Not perfect so open to any suggestions :)

### CODE REVIEW

File: local/README.md
Comment (aecay) [inline local/README.md:38]: Would it make sense to use a docker mount to pull in the journeys, so that we don't need a new image if they change?  Something like https://docs.docker.com/storage/bind-mounts/#start-a-container-with-a-bind-mount (modified to fit into docker-compose).  This is not something that needs to be done for this PR, but a future suggestion of how to reduce the number of image rebuilds that are needed while testing code.

(Hard mode: the runner imports the journeys when it starts [or at latest when it runs the journey for the first time], and probably does not re-import them on change.  So the runners will need a restart when the journeys change.  To do that without a full image rebuild, the runner probably needs to watch the journey files with inotify or some such, and exit if it sees a change there...)

**Review by aecay — APPROVED** [review]
This looks great!  Just one suggestion for future improvements if you keep working on this, but I think it's already a massive addition :+1:

File: local/README.md
Comment (haffi96) [reply local/README.md:38]: That would be pretty nice, will take a look at those. Thanks!

Also, not sure what's going on with the CI checks 🤔

**Review by Matt2298 — APPROVED** [review]
Looks like some great improvements!

**Comment by haffi96** [comment]
I think im going to need some help with the CI checks. I can't click on them and see the status cause jenkins perms 🤷

**Comment by aecay** [comment]
I'm stumped -- the CI jobs are running for other PRs, but apparently not this one.  I can't see a reason for that.  I now lack perms to trigger the jobs on ID's jenkins (:crying_cat_face:).  For someone who does -- I think manually triggering the [_trigger job](https://jenkins-identity.tools.cosmic.sky/job/identity/job/mite/job/_trigger/) against the latest commit on this branch should do the trick....

**Comment by haffi96** [comment]
If that doesn't work^. It could be something to do with the fact that this PR is from a forked repo 😕

**Comment by aecay** [comment]
> this PR is from a forked repo

oh yep, that's it.  do you still have perms to this repo?  imo you should do, although apparently sky restricts the permission to add collaborators from outside the organization to company-level github admins.  :confounded: 

we (= @jb098 as head maintainer) should decide whether to:
- look for a better/more open CI soln than core jenkins [which itself might involve interaction with sky's GH admins for various reasons]
- go to the admins and get you added to this repo as a collaborator (and ideally have a way to do this in the future as a matter of course...one step at time i guess)

on a pragmatic basis, i (or someone) can copy the branch from your fork into this repo so we can run the ci and merge it, though this isn't really a good long-term solution... (if you want to proceed in this way, i can make the copy this evening if no one beats me to it...)

**Comment by haffi96** [comment]
@aecay Yep, that sounds like a good solution for now. Thanks Aaron! Obvs there's no hurry to get this merged, so take your time

**Comment by aecay** [comment]
I've created the branch and a PR: https://github.com/sky-uk/mite/pull/233

**Comment by aecay** [comment]
Weirdly I think the CI on that branch will actually mark this PR green -- because the marking is by commit, and the commits in both branches are identical.  So if that works, I'll close the other PR and we can merge from this one (in fact, pushing the branch would have worked without even my creating a PR...the more you know....)

**Comment by aecay** [comment]
Black is unhappy:

```
[2022-02-17T12:23:17.658Z] diff --git a/local/demo.py b/local/demo.py
[2022-02-17T12:23:17.659Z] index 4eab9a5..2c69ad6 100644
[2022-02-17T12:23:17.659Z] --- a/local/demo.py
[2022-02-17T12:23:17.659Z] +++ b/local/demo.py
[2022-02-17T12:23:17.659Z] @@ -1,15 +1,17 @@
[2022-02-17T12:23:17.659Z] -from mite_http import mite_http
[2022-02-17T12:23:17.659Z]  from mite.scenario import StopScenario
[2022-02-17T12:23:17.659Z] +from mite_http import mite_http
[2022-02-17T12:23:17.659Z]  
[2022-02-17T12:23:17.659Z]  
[2022-02-17T12:23:17.659Z]  def volume_model_factory(n):
[2022-02-17T12:23:17.659Z]      def vm(start, end):
[2022-02-17T12:23:17.659Z] -        if start > 60 * 15: # Will run for 15 mins
[2022-02-17T12:23:17.659Z] +        if start > 60 * 15:  # Will run for 15 mins
[2022-02-17T12:23:17.659Z]              raise StopScenario
[2022-02-17T12:23:17.659Z]          return n
[2022-02-17T12:23:17.659Z] +
[2022-02-17T12:23:17.659Z]      vm.__name__ = f"volume model {n}"
[2022-02-17T12:23:17.659Z]      return vm
[2022-02-17T12:23:17.659Z]  
[2022-02-17T12:23:17.659Z] +
[2022-02-17T12:23:17.659Z]  # Mock Server
[2022-02-17T12:23:17.659Z]  _API_URL = "http://apiserver:8000/"  # apiserver is for docker-compose. Change this to http://localhost:8000/[](https://jenkins-identity.tools.cosmic.sky/blue/organizations/jenkins/identity%2Fmite%2Fmite-ci/detail/mite-ci/986/pipeline#step-37-log-53) for local testing
[2022-02-17T12:23:17.659Z]  # TODO make http://apiserver:8000/[](https://jenkins-identity.tools.cosmic.sky/blue/organizations/jenkins/identity%2Fmite%2Fmite-ci/detail/mite-ci/986/pipeline#step-37-log-54) env var for controller and runners and default _API_URL to localhost
[2022-02-17T12:23:17.659Z] +
```

**Comment by aecay** [comment]
> Now to trigger the CI again, are you going to have to make another push

done.  il n y'a pas de probleme

**Comment by haffi96** [comment]
Nice, checks passed too. Thankyou 🙌  Only problem is I don't have access to merge haha

**Pull request merged**
